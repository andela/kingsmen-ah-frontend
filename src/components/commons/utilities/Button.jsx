import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import '../index.scss';

function Button({ type, children, color, stretch, onClick, ...rest }) {
  return (
    <button
      type='submit'
      className={classnames(
        'bg-transparent hover:text-white py-2 px-4 border  rounded mr-2 text-sm',
        {
          [`hover:bg-${color}-700 text-${color}-700 border-${color}-700 hover:border-transparent ${
            stretch ? 'block w-full' : ''
          }`]: type === 'outlined'
        },
        {
          [`text-white bg-${color}-700 hover:bg-${color}-800 ${
            stretch ? 'block w-full' : ''
          }`]: type === 'solid'
        },
        {
          [`text-${color}-700 hover:text-${color}-800 border-transparent`]:
            type === 'regular'
        }
      )}
      onClick={onClick}
      {...rest}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.element, PropTypes.string
  ]).isRequired,
  color: PropTypes.string.isRequired,
  stretch: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  stretch: false
};

export default Button;
