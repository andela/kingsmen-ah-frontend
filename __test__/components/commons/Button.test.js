import React from 'react';
import Button from '@components/commons/utilities/Button';

const props = {
  type: 'solid',
  color: 'blue',
  onClick: jest.fn()
};

const setup = () => {
  const wrapper = shallow(<Button {...props}>Login</Button>);
  return wrapper;
};

describe('<Button />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the Button component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Button')).toBeTruthy();
  });

  it('should display Login as button text', () => {
    expect(wrapper.props().children).toEqual('Login');
  });

  it('should pass since button exists', () => {
    expect(wrapper).toHaveLength(1);
  });
});
