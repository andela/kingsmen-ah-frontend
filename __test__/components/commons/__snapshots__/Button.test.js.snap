// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the Button component correctly 1`] = `
<button
  className="bg-transparent hover:text-white py-2 px-4 border  rounded mr-2 text-sm text-white bg-blue-700 hover:bg-blue-800 "
  onClick={[MockFunction]}
  type="submit"
>
  Login
</button>
`;
